compiler/fir/java/src/org/jetbrains/kotlin/fir/java/JavaUtils.kt

```diff
@@ -421,52 +428,120 @@ private fun FirRegularClass.createRawArguments(
+        buildEnumCall(session, ClassId.topLevel(StandardNames.FqNames.annotationRetention), Name.identifier(it.name))
+    }
+
+private fun buildArgumentMapping(
     session: FirSession,
     javaTypeParameterStack: JavaTypeParameterStack,
-    classId: ClassId?,
+    lookupTag: ConeClassLikeLookupTagImpl,
     annotationArguments: Collection<JavaAnnotationArgument>
-): LinkedHashMap<FirExpression, FirValueParameter>? {
-    if (classId == null) {
-        annotationTypeRef = buildErrorTypeRef { diagnostic = ConeUnresolvedReferenceError() }
+): FirArgumentList? {
+    if (annotationArguments.none { it.name != null }) {
         return null
     }
-    val lookupTag = ConeClassLikeLookupTagImpl(classId)
-    annotationTypeRef = buildResolvedTypeRef {
-        type = ConeClassLikeTypeImpl(lookupTag, emptyArray(), isNullable = false)
-    }
-    if (annotationArguments.any { it.name != null }) {
-        val mapping = linkedMapOf<FirExpression, FirValueParameter>()
-        val annotationClassSymbol = session.firSymbolProvider.getClassLikeSymbolByFqName(classId).also {
-            lookupTag.bindSymbolToLookupTag(session, it)
-        }
-        if (annotationClassSymbol != null) {
-            val annotationConstructor =
-                (annotationClassSymbol.fir as FirRegularClass).declarations.filterIsInstance<FirConstructor>().first()
-            for (argument in annotationArguments) {
-                mapping[argument.toFirExpression(session, javaTypeParameterStack)] =
-                    annotationConstructor.valueParameters.find { it.name == (argument.name ?: JavaSymbolProvider.VALUE_METHOD_NAME) }
-                        ?: return null
-            }
-            return mapping
-        }
+    val annotationClassSymbol = session.firSymbolProvider.getClassLikeSymbolByFqName(lookupTag.classId).also {
+        lookupTag.bindSymbolToLookupTag(session, it)
+    } ?: return null
+    val annotationConstructor =
+        (annotationClassSymbol.fir as FirRegularClass).declarations.filterIsInstance<FirConstructor>().first()
+    val mapping = annotationArguments.associateTo(linkedMapOf()) { argument ->
+        val parameter = annotationConstructor.valueParameters.find { it.name == (argument.name ?: JavaSymbolProvider.VALUE_METHOD_NAME) }
+            ?: return null
+        argument.toFirExpression(session, javaTypeParameterStack) to parameter
     }
-    return null
+    return buildResolvedArgumentList(mapping)
 }
 
 internal fun JavaAnnotation.toFirAnnotationCall(
     session: FirSession, javaTypeParameterStack: JavaTypeParameterStack
 ): FirAnnotationCall {
     return buildAnnotationCall {
-        val mapping = buildArgumentMapping(session, javaTypeParameterStack, classId, arguments)
-        argumentList = if (mapping != null) {
-            buildResolvedArgumentList(mapping)
+        val lookupTag = when (classId) {
+            JAVA_TARGET_CLASS_ID -> ClassId.topLevel(StandardNames.FqNames.target)
```

Comment:
You can use `JavaAnnotationMapper.javaToKotlinNameMap` access here, instead of writing this `when`

Reply:
JavaAnnotationMapper is in the core/descriptors.jvm module, so it's not currently listed as a dependency, and probably shouldn't be.